src/v/utils/base64.cc

```diff
@@ -110,3 +120,25 @@ ss::sstring iobuf_to_base64(const iobuf& input) {
     output.resize(written);
     return output;
 }
+
+bytes base64url_to_bytes(std::string_view data) {
+    bytes rv{bytes::initialized_later{}, data.size()};
+    int bits_collected = -8;
+    int accumulator = 0;
+    int pos = 0;
+    for (const unsigned char c : data) {
+        if (c > 127 || decode_url_table[c] < 0) {
+            throw base64_url_decoder_exception();
+        }
+        accumulator = (accumulator << 6) + decode_url_table[c];
+        bits_collected += 6;
+        if (bits_collected >= 0) {
+            auto val = static_cast<bytes::value_type>(
+              (accumulator >> bits_collected) & 0xff);
+            rv[pos++] = val;
+            bits_collected -= 8;
+        }
```

Comment:
Do we need to reset the `accumulator` here as well by shifting it right? Otherwise, I'm wondering if it would overflow.